compose/service.py

```diff
@@ -1069,6 +1071,28 @@ def parse_restart_spec(restart_config):
 
     return {'Name': name, 'MaximumRetryCount': int(max_retry_count)}
 
+# Ulimits
+
+
+def build_ulimits(ulimit_config):
+    if not ulimit_config:
+        return None
+    ulimits = []
+    for limit_name, soft_hard_values in six.iteritems(ulimit_config):
+        if isinstance(soft_hard_values, six.integer_types):
+            ulimits.append({'name': limit_name, 'soft': soft_hard_values, 'hard': soft_hard_values})
+        elif isinstance(soft_hard_values, dict):
+            if not soft_hard_values['soft'] <= soft_hard_values['hard']:
+                raise ConfigError(
```

Comment:
Just to make sure I start down the right path, you're saying adding that validation by calling a validate_ulimits function somewhere around https://github.com/docker/compose/blob/master/compose/config/config.py#L348, correct?

Is the ultimate goal to factor out all of the ConfigErrors from service.py?